frontend/src/components/TableRatedTaskRow.tsx

```diff
@@ -0,0 +1,84 @@
+import React, { useState } from 'react';
+import classNames from 'classnames';
+import { Link } from 'react-router-dom';
+import ArrowForwardIcon from '@material-ui/icons/ArrowForward';
+import DoneAllIcon from '@material-ui/icons/DoneAll';
+import { Trans } from 'react-i18next';
+import { Task } from '../redux/roadmaps/types';
+import css from './TableRatedTaskRow.module.scss';
+import { paths } from '../routers/paths';
+import { calcTaskValues, calcTaskWorks } from '../utils/TaskUtils';
+
+const classes = classNames.bind(css);
+
+interface TableTaskRowRatedProps {
+  task: Task;
+}
+
+export const TableRatedTaskRow: React.FC<TableTaskRowRatedProps> = ({
+  task,
+}) => {
+  const [hovered, setHovered] = useState(false);
+
+  const numFormat = (num: number) => {
+    if (Number.isNaN(num)) return 0;
+    return num.toLocaleString(undefined, {
+      minimumFractionDigits: 0,
+      maximumFractionDigits: 1,
+    });
+  };
+
+  return (
+    <tr
+      className={classes(css.tableRow, {
+        [css.hoverRow]: hovered,
+        [css.completedRow]: task.completed,
+      })}
+    >
+      <td className={classes(css.ratedTitle)}>
+        {task.completed && <DoneAllIcon className={classes(css.doneIcon)} />}
+        {task.name}
+      </td>
+      <td className={classes(css.ratedTd)}>
+        {numFormat(calcTaskValues(task).sum / calcTaskValues(task).count)}
+      </td>
```

Comment:
I'm a bit uneasy with the amount of duplicate calls here

What I was suggesting earlier by doing the calculation once (sorry if I was unclear):

> Calculating all the necessary averages and sums could be done in one pass over the task ratings

> could also be used in the `TableRatedTaskRow` directly, instead of all the `calc*` calls

Something like this could work:

```suggestion
  const totalRatings = totalRatingsByDimension(task);
  const noRatings = { sum: 0, count: 1 };
  const valueRatings =
    totalRatings.get(TaskRatingDimension.BusinessValue) ?? noRatings;
  const workRatings =
    totalRatings.get(TaskRatingDimension.RequiredWork) ?? noRatings;

  return (
    <tr
      className={classes(css.tableRow, {
        [css.hoverRow]: hovered,
        [css.completedRow]: task.completed,
      })}
    >
      <td className={classes(css.ratedTitle)}>
        {task.completed && <DoneAllIcon className={classes(css.doneIcon)} />}
        {task.name}
      </td>
      <td className={classes(css.ratedTd)}>
        {numFormat(valueRatings.sum / valueRatings.count)}
```